### ISSUE TITLE
BCD note  layout is broken in visibilitychange event page

### ISSUE THREAD

**Issue by wbamberg** [mdn/yari#3076]
Visit https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event#browser_compatibility and click the disclosure arrow for Safari, which is showing partial support:

<img width="526" alt="Screen Shot 2021-02-26 at 2 00 48 PM" src="https://user-images.githubusercontent.com/432915/109359925-23906000-783b-11eb-892e-096a9c30e9bc.png">

**Comment by Ryuno-Ki** [comment]
Looking into the inspector, I can see a wrap inside a `<dl>` which gets assigned a `max-width`.

![safari-broken](https://user-images.githubusercontent.com/3097194/109362775-03d55800-788c-11eb-941a-d26c15e29aa2.png)

That's where I would start looking into.

**Comment by tannerdolby** [comment]
@peterbe Sure thing. Thanks for the heads up.

**Comment by tannerdolby** [comment]
The problem stemmed from the `.bc-history .bc-notes-wrapper` container where if  a `<dt>` had more than one adjacent `<dl>` sibling then they would continue growing in a row until the text overflows parent `<tr>`. 

<img width="901" alt="Screen Shot 2021-04-21 at 1 55 14 AM" src="https://user-images.githubusercontent.com/48612525/115526420-de8a1800-a244-11eb-9849-db39164c502a.png">

Since all of the `.bc-notes-wrapper` divs are flex containers, they are ensuring the nested `<dt>` and `<dl>` elements are placed into one row and text that overflows each `<dd>` content box will flow to a new line, which is why the first box is so tall. This works great when a `<dt>` only has a single adjacent `<dl>` as the text will wrap onto the next line once it reaches `85ch` from `max-width: 85ch`.

```html
<div class="bc-notes-wrapper">
  <dt class="bc-supports-partial bc-supports">
  <dd>...</dd>
  <dd>...</dd>
  <dd>...</dd>
</div>
```
 

For the majority of `<div class="bc-notes-wrapper>` containers, each one only has a single adjacent`<dd>` element per `<dt class="bc-supports-yes">`. Since the partial support dropdowns in this BCD table have three adjacent `<dd>` elements for one `<dt>`, the flex row layout is simply stretching each of the content boxes so they all fit in a row.  The `max-width` properties do control the amount of space amount of space each `<dl>` can occupy, so if we remove the `max-width` from `.main-page-content dl` and `.bc-history dl` it looks like:

<img width="940" alt="Screen Shot 2021-04-21 at 3 51 41 AM" src="https://user-images.githubusercontent.com/48612525/115542231-ec479980-a254-11eb-96de-bbe2d1e86086.png">

**Comment by tannerdolby** [comment]
### Potential fix

Adding `word-break: break-word;` to each nested `<dd>` seems to be one potential fix. The `max-width: 85ch` from `main-page-content dl` still makes the the three `<dd>` elements look crammed together, but this is somewhat of an improvement and works all the way down to media query that triggers `display: none`.

<img width="908" alt="Screen Shot 2021-04-21 at 4 59 49 AM" src="https://user-images.githubusercontent.com/48612525/115550399-ab548280-a25e-11eb-810e-78ea5c663499.png">


OR

Using a `grid` layout, things look pretty nice and nothing breaks. I wasn't entirely sure how the three adjacent `<dd>` elements should be laid out for a single `<dt>`, so I imagined they would be stacked in a column one after another since they would overflow out the right most likely if we tried to squeeze all three of them into a single row. Looks good all the way until `display: none` comes at smaller viewport widths.

```css
.bc-notes-wrapper {
  display: grid;
  grid-template-columns: repeat(6, 1fr);
}
```

<img width="904" alt="Screen Shot 2021-04-21 at 1 29 02 AM" src="https://user-images.githubusercontent.com/48612525/115522605-1d1dd380-a241-11eb-8750-f72b6f908926.png">

**Comment by peterbe** [comment]
@schalkneethling can understand 100% of what you wrote. Me, about 2%. :)

**Comment by tannerdolby** [comment]
@peterbe ok :) I will wait to hear back from Schalk. Out of the two "potential fix" layouts mentioned in my previous comment, which one do you think looks better? Also in both of the videos included, you can see #2692 towards the end.

Option one (CSS Grid): have the the `.bc-notes-wrapper` become a grid and then position the `<dd>` elements so they span grid columns 2-6 and stack in a column layout. I think this looks good in terms of occupying space.

https://user-images.githubusercontent.com/48612525/115639903-931a4d00-a2ca-11eb-82bc-882bf1302769.mov

Option two: using `word-break` on  each [`<dd>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dd) element to provide a little nicer layout. (still doesn't look great as things are squished from the parent containers `max-width: 85ch` declaration).

https://user-images.githubusercontent.com/48612525/115639300-1aff5780-a2c9-11eb-9816-d13909e94dbb.mov

**Comment by tannerdolby** [comment]
> Isn't hands-down 10x better to interact with as described by the first video

It is or isn't hands down better in first video? I like the first videos layout better but I know @schalkneethling mentioned he was working on something to fix this while we were discussing #2692.

Yeah, I agree the text is hopeless in video 2, that means using `word-break: break-word` on the `<dd>` tags is out the window. Therefore, my vote goes with the implementation in video 1 or whatever layout Schalk spins up.

**Comment by peterbe** [comment]
> It is or isn't hands down better in first video? I like the first videos layout better but I know @schalkneethling mentioned he was working on something to fix this while we were discussing #2692.

Sorry if I wasn't clear. From a reader's point of view video1 is much better than video2.

**Comment by tannerdolby** [comment]
@peterbe No worries, thanks for clarifying.
